### PR TITLE
Add Morpho FlashLoan Callback

### PR DESCRIPTION
**Pull request by Jeff-CCH**
- [x] flash loan callback
- [x] test
- [x] deployment script

### CODE REVIEW

File: test/integration/Morpho.t.sol [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import {Test} from 'forge-std/Test.sol';
+import {IERC20} from 'lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol';
+import {Router, IRouter} from 'src/Router.sol';
+import {IAgent} from 'src/interfaces/IAgent.sol';
+import {DataType} from 'src/libraries/DataType.sol';
+import {Test} from 'forge-std/Test.sol';
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:8]: Duplicate import

File: test/integration/Morpho.t.sol [hunk 2]
```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import {Test} from 'forge-std/Test.sol';
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:3]: Give one blank line after pragma

File: test/integration/Morpho.t.sol [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import {DataType} from 'src/libraries/DataType.sol';
+import {Test} from 'forge-std/Test.sol';
+import {MorphoFlashLoanCallback} from 'src/callbacks/MorphoFlashLoanCallback.sol';
+
+interface IMorphoFlashLoanCallback {
+    function flashLoan(address token, uint256 assets, bytes calldata data) external;
+}
+
+contract MorphoIntegrationTest is Test {
+    // using SafeERC20 for IERC20;
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:16]: Do we still need this line?

File: test/integration/Morpho.t.sol [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        vm.label(address(router), 'Router');
+        vm.label(address(agent), 'Agent');
+        vm.label(MORPHO, 'Morpho');
+        vm.label(address(flashLoanCallback), 'MorphoFlashLoanCallback');
+        vm.label(address(USDC), 'USDC');
+    }
+
+    function testExecuteMorphoFlashLoan(uint256 amount) external {
+        IERC20 borrowedToken = USDC;
+        address flashloanPool = MORPHO;
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:49]:
```suggestion
        address flashLoanPool = MORPHO;
```
However, I suggest to use `MORPHO` directly as all tokens are in this singleton contract even we change `USDC` to other token afterward.

File: test/integration/Morpho.t.sol [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import {Router, IRouter} from 'src/Router.sol';
+import {IAgent} from 'src/interfaces/IAgent.sol';
+import {DataType} from 'src/libraries/DataType.sol';
+import {Test} from 'forge-std/Test.sol';
+import {MorphoFlashLoanCallback} from 'src/callbacks/MorphoFlashLoanCallback.sol';
+
+interface IMorphoFlashLoanCallback {
+    function flashLoan(address token, uint256 assets, bytes calldata data) external;
+}
+
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:14]: Can use protocolink-contract/src/interfaces/morphoblue/IMorpho.sol directly.
```suggestion
import {MorphoFlashLoanCallback, IMorphoFlashLoanCallback, IMorpho} from 'src/callbacks/MorphoFlashLoanCallback.sol';
```

File: test/integration/Morpho.t.sol [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+contract MorphoIntegrationTest is Test {
+    // using SafeERC20 for IERC20;
+
+    address public constant MORPHO = 0x64c7044050Ba0431252df24fEd4d9635a275CB41;
+    IERC20 public constant USDC = IERC20(0x62bD2A599664D421132d7C54AB4DbE3233f4f0Ae);
+
+    address public user;
+    IRouter public router;
+    IAgent public agent;
+    MorphoFlashLoanCallback public flashLoanCallback;
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:24]:
```suggestion
    IMorphoFlashLoanCallback public flashLoanCallback;
```

File: test/integration/Morpho.t.sol [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    function _logicMorphoFlashLoan(address token, uint256 amount) public view returns (DataType.Logic memory) {
+        // Encode logic
+        bytes memory params = _encodeExecute(token, amount);
+
+        return
+            DataType.Logic(
+                address(flashLoanCallback), // to
+                abi.encodeWithSelector(MorphoFlashLoanCallback.flashLoan.selector, token, amount, params),
```
Comment (zodahu) [inline test/integration/Morpho.t.sol:75]:
```suggestion
                abi.encodeWithSelector(IMorpho.flashLoan.selector, token, amount, params),
```

**Review by zodahu — APPROVED** [review]
LGTM

**Review by hihiben — APPROVED** [review]
LGTM

**Pull request merged**